### PR TITLE
Ins 3293 goroutine leak

### PR DESCRIPTION
**Pull request by SergeyRadist**
#INS-3293 leaking goroutins fix  
- added force termination

### CODE REVIEW

File: actor/system/system.go [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
 func New() actor.System {
 	return &system{
-		mailboxes: make(map[actor.Pid] mailbox.Mailbox),
+		mailboxes: make(map[actor.Pid]mailbox.Mailbox),
 	}
 }
 
 func actorLoop(actor actor.Actor, mailbox mailbox.Mailbox) {
-	var err error
+	// var err error
```
Comment (aromancev) [inline actor/system/system.go:26]: ?

File: actor/system/system.go [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
 	for {
 		prevState := actor
-		message := mailbox.Dequeue()
+		message, err := mailbox.Dequeue()
+		if err != nil && err == errors.Terminate {
```
Comment (aromancev) [inline actor/system/system.go:30]: Handle other errors.

File: actor/system/system.go [hunk 3]
```diff
@@ -98,3 +102,9 @@ func (s *system) SendPriority(pid actor.Pid, message actor.Message) error {
 func (s *system) AwaitTermination() {
 	s.wg.Wait()
 }
+
+func (s *system) CloseAll() {
+	for _, mb := range s.mailboxes {
```
Comment (aromancev) [inline actor/system/system.go:107]: lock?

**Review by aromancev — CHANGES_REQUESTED** [review]

File: actor/system/system.go (hunk 3 above)
Comment (SergeyRadist) [reply actor/system/system.go:107]: added

File: actor/mailbox/mailbox.go [hunk 4]
```diff
@@ -108,26 +112,30 @@ func (mb *mailbox) Unstash() {
 	// that is waiting in Dequeue.
 }
 
-func (mb *mailbox) Dequeue() actor.Message {
-	var msg actor.Message
+func (mb *mailbox) Dequeue() (actor.Message, error) {
 	for {
 		mb.lock.Lock()
```
Comment (ita-sammann) [inline actor/mailbox/mailbox.go:117]: Let's add closure here and `defer mb.lock.Unlock()`

**Pull request closed without merging**